locopy/snowflake.py

```diff
@@ -424,20 +424,36 @@ def insert_dataframe_to_table(
             If metadata==None, it will be generated based on data
         """
         import pandas as pd
+        import polars as pl
+        import polars.selectors as cs
 
         if columns:
-            dataframe = dataframe[columns]
+            try:
+                dataframe = dataframe[columns]
+            except TypeError:
+                dataframe = dataframe.select(columns)
 
         all_columns = columns or list(dataframe.columns)
         column_sql = "(" + ",".join(all_columns) + ")"
         string_join = "(" + ",".join(["%s"] * len(all_columns)) + ")"
 
         # create a list of tuples for insert
         to_insert = []
-        for row in dataframe.itertuples(index=False):
-            none_row = tuple(None if pd.isnull(val) else str(val) for val in row)
-            to_insert.append(none_row)
-
+        if isinstance(dataframe, pd.DataFrame):
+            for row in dataframe.itertuples(index=False):
+                none_row = tuple(None if pd.isnull(val) else str(val) for val in row)
+                to_insert.append(none_row)
+        elif isinstance(dataframe, pl.DataFrame):
```

Comment:
Is the only difference here between `DataFrame` and `LazyFrame` the `collect`? If so, I think we should consider just supporting DataFrame and users collect before inserting. For example, the polars API has `sink_csv` for LazyFrame, and `write_csv` for DataFrame but only `write_delta` for DataFrame. I think it's ok in this case to just support dataframe if we explicitly have to collect in the method and we're not doing any intense manipulation beyond selecting columns and filling with NaN.

Reply:
Yes, the only difference is just the `collect` in this `insert_dataframe_to_table` method. On a second thought, we could still support both lazyframe and dataframe by calling `.lazy()` if it is either `polars.DataFrame` or `polars.LazyFrame` and have a single block of code instead of two, even though calling `.lazy()` on lazyframe doesn't make much sense, it simplifies the code and still provides support for both

Reply:
I think it's best to only support `DataFrame`. Because we're not using any lazy functionality specifically, having a hidden collect can be a gotcha for users using LazyFrames where they may be hit with penalties for having collect in here and then collect later on. If we're explicit about DataFrame only, then users can optimize when to collect themselves.

Reply:
That's a fair point, down to drop support for LazyFrame.